imucapture/src/main.cpp

```diff
@@ -40,29 +41,116 @@
 #include <iomanip>
 #include <stdexcept>
 #include <string>
-#include <fstream>  //std::ofstream
+#include <fstream>
 #include <ctime>
 
 #include <cstdio>
+#include <stdlib.h>
 #include <termios.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/time.h>
 
-void changemode(int dir){
+void changemode(int dir);
+int kbhit();
+
+void device_initialization(DeviceClass &device, XsPortInfo &mtPort);
+XsPortInfo scan_usb_devices();
+
+void set_message_to_packet(
+	XsDataPacket &packet,
+	XsMessage &msg,
+	XsDeviceId dev_id);
+
+inline void
+print_position_to_file(std::ofstream &outfile, XsDataPacket &packet);
+
+inline void
+print_quaternion_to_file(std::ofstream &outfile, XsDataPacket &packet);
+
+inline void
+print_euler_to_file(std::ofstream &outfile, XsDataPacket &packet);
+
+inline void
+print_timestamp_to_file(std::ofstream &outfile, XsDataPacket &packet);
+
+int main(int argc, char* argv[])
+{
+	DeviceClass device;
+	auto start = clock();
+	XsPortInfo mtPort;
+	mtPort = scan_usb_devices();
+
+	try {
+		device_initialization(device, mtPort);
+	}
+	catch (std::runtime_error const & error) {
+		std::cout << error.what() << std::endl;
+		exit(EXIT_FAILURE);
+	}
+	catch (...) {
+		std::cout << "An unknown fatal error has occured. "
+			<< "Aborting." << std::endl;
+		exit(EXIT_FAILURE);
+	}
+
+	XsByteArray data;
+	XsMessageArray msgs;
+
+	while (!kbhit()) {
+		//TODO:Move to chrono
+		auto duration = (clock() - start) / CLOCKS_PER_SEC;
+
+		device.readDataToBuffer(data);
+		device.processBufferedData(data, msgs);
+		for (auto msg : msgs) {
+			
+			XsDataPacket packet;
+			set_message_to_packet(packet, msg, mtPort.deviceId());
+			
+			std::ofstream imu_txt;
+			imu_txt.open("IMU.txt", std::ofstream::out | std::ofstream::app);
+			
+			print_position_to_file(imu_txt, packet);
+			print_quaternion_to_file(imu_txt, packet);
+			print_euler_to_file(imu_txt, packet);
+			print_timestamp_to_file(imu_txt, packet);
+
+			imu_txt << std::endl;
+			imu_txt.close();
+		}
+		msgs.clear();
+		XsTime::msleep(0);
+	}
+
+	std::cout << "Closing port..." << std::endl;
+	device.close();
+
+	std::cout << "Successful exit." << std::endl;
+
+	return 0;
+}
+
+
+
+
+
+void changemode(int dir)
+{
     static struct termios oldt, newt;
 
-    if (dir == 1){
+    if (dir == 1) {
         tcgetattr( STDIN_FILENO, &oldt);
         newt = oldt;
         newt.c_lflag &= ~( ICANON | ECHO );
         tcsetattr( STDIN_FILENO, TCSANOW, &newt);
-    }else{
+    } else {
         tcsetattr( STDIN_FILENO, TCSANOW, &oldt);
 	}
 }
 
-int kbhit (){
+int kbhit ()
```

Comment:
Could this be inlined?

Reply:
The function makes use of a few OS calling functions, so it may not be able to be inlined, although, that's just my guess. If anything, we can just make it inline anyway and have the compiler decide whether it's needed or not.